service/bfg/bfg.go

```diff
@@ -284,6 +280,10 @@ func (s *Server) handleRequest(parentCtx context.Context, bws *bfgWs, wsid strin
 
 	log.Tracef("Handling request %v: %v", bws.addr, cmd)
 
+	// ensure that only one handler is called per-time-per-connection
+	bws.mtx.Lock()
```

Comment:
hmm now that I think about it, this would cause longer running requests to block shorter ones, need to think a bit more 🤔